test/mean_function.jl

```diff
@@ -1,47 +1,53 @@
 @testset "mean_functions" begin
     @testset "ZeroMean" begin
-        P = 3
-        Q = 2
-        D = 4
-        # X = ColVecs(randn(rng, D, P))
-        x = randn(P)
-        x̄ = randn(P)
-        f = ZeroMean{Float64}()
+        rng, N, D = MersenneTwister(123456), 5, 3
+        x1 = randn(rng, N)
+        xD = ColVecs(randn(rng, D, N))
+        xD′ = RowVecs(randn(rng, N, D))
 
-        for x in [x]
-            @test AbstractGPs._map_meanfunction(f, x) == zeros(size(x))
-            # differentiable_mean_function_tests(f, randn(rng, P), x)
-        end
+        m = ZeroMean{Float64}()
+
+        for x in [x1, xD, xD′]
+            @test AbstractGPs._map_meanfunction(m, x) == zeros(size(x))
+            differentiable_mean_function_tests(m, randn(rng, N), x)
 
-        # Manually verify the ChainRule. Really, this should employ FiniteDifferences, but
-        # currently ChainRulesTestUtils isn't up to handling this, so this will have to do
-        # for now.
-        y, pb = rrule(AbstractGPs._map_meanfunction, f, x)
-        @test y == AbstractGPs._map_meanfunction(f, x)
-        Δmap, Δf, Δx = pb(randn(P))
-        @test iszero(Δmap)
-        @test iszero(Δf)
-        @test iszero(Δx)
+            # Manually verify the ChainRule. Really, this should employ FiniteDifferences, but
+            # currently ChainRulesTestUtils isn't up to handling this, so this will have to do
+            # for now.
+            y, pb = rrule(AbstractGPs._map_meanfunction, m, x)
+            @test y == AbstractGPs._map_meanfunction(m, x)
+            Δmap, Δf, Δx = pb(randn(rng, N))
+            @test iszero(Δmap)
+            @test iszero(Δf)
+            @test iszero(Δx)
+        end
     end
     @testset "ConstMean" begin
-        rng, D, N = MersenneTwister(123456), 5, 3
-        # X = ColVecs(randn(rng, D, N))
-        x = randn(rng, N)
+        rng, N, D = MersenneTwister(123456), 5, 3
+        x1 = randn(rng, N)
+        xD = ColVecs(randn(rng, D, N))
+        xD′ = RowVecs(randn(rng, N, D))
+
         c = randn(rng)
         m = ConstMean(c)
 
-        for x in [x]
+        for x in [x1, xD, xD′]
             @test AbstractGPs._map_meanfunction(m, x) == fill(c, N)
-            # differentiable_mean_function_tests(m, randn(rng, N), x)
+            differentiable_mean_function_tests(m, randn(rng, N), x)
         end
     end
     @testset "CustomMean" begin
-        rng, N, D = MersenneTwister(123456), 11, 2
-        x = randn(rng, N)
+        rng, N, D = MersenneTwister(123456), 5, 3
+        x1 = randn(rng, N)
+        xD = ColVecs(randn(rng, D, N))
+        xD′ = RowVecs(randn(rng, N, D))
+
         foo_mean = x -> sum(abs2, x)
-        f = CustomMean(foo_mean)
+        m = CustomMean(foo_mean)
```

Comment:
Also unrelated it seems:
```suggestion
        f = CustomMean(foo_mean)
```

Reply:
again, it was `m` in some places and `f` in others, and I wanted to make it consistent. `m` seemed to make more sense to me (it's not a function, can't be called as it is at the moment)